core/src/main/java/com/linecorp/armeria/client/DefaultWebClient.java

```diff
@@ -44,36 +39,32 @@
 
     @Override
     public HttpResponse execute(HttpRequest req) {
-        URI uri;
-
-        if (isAbsoluteUri(req.path())) {
-            try {
-                uri = URI.create(req.path());
-            } catch (Exception ex) {
-                logger.warn("Failed to create URI: {}", req.path(), ex);
-                uri = null;
-            }
-        } else {
-            if (req.scheme() != null && req.authority() != null) {
+        if (Clients.isUndefinedUri(uri())) {
+            final URI uri;
+            if (isAbsoluteUri(req.path())) {
+                try {
+                    uri = URI.create(req.path());
+                } catch (Exception ex) {
+                    return failureResponse(req, new IllegalArgumentException(
+                            "Failed to create a URI: " + req.path(), ex));
+                }
+            } else if (req.scheme() != null && req.authority() != null) {
                 uri = req.uri();
             } else {
-                uri = null;
+                return failureResponse(req, new IllegalArgumentException("no authority: " + req.path()));
```

Comment:
How about `"A URI with scheme and authority must be specified: "`?

Reply:
Fixed. 😉